src/masternodes/mn_checks.cpp

```diff
@@ -2943,6 +2947,59 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         return mnview.UpdateVault(obj.vaultId, *vault);
     }
 
+    Res CollateralPctCheck(const bool hasDUSDLoans, const CCollateralLoans& collateralsLoans, const uint32_t ratio) const {
+
+        std::optional<std::pair<DCT_ID, std::optional<CTokensView::CTokenImpl>>> tokenDUSD;
+        if (static_cast<int>(height) >= consensus.FortCanningRoadHeight) {
+            tokenDUSD = mnview.GetToken("DUSD");
+        }
+
+        // Calculate DFI and DUSD value separately
+        uint64_t totalCollateralsDUSD = 0;
+        uint64_t totalCollateralsDFI = 0;
+
+        for (auto& col : collateralsLoans.collaterals){
+            if (col.nTokenId == DCT_ID{0} )
+                totalCollateralsDFI += col.nValue;
+
+            if(tokenDUSD && col.nTokenId == tokenDUSD->first)
+                totalCollateralsDUSD += col.nValue;
+        }
+        auto totalCollaterals = totalCollateralsDUSD + totalCollateralsDFI;
+
+        // Heigh checks
+        auto isPostFCH = static_cast<int>(height) >= consensus.FortCanningHillHeight;
+        auto isPreFCH = static_cast<int>(height) < consensus.FortCanningHillHeight;
+        auto isPostFCE = static_cast<int>(height) >= consensus.FortCanningEpilogueHeight;
+        auto isPostFCR = static_cast<int>(height) >= consensus.FortCanningRoadHeight;
+
+        // Condition checks
+        auto isDFIHalfOfAllCollaterall = totalCollateralsDFI < collateralsLoans.totalCollaterals / 2;
+        auto isDFIAndDUSDHalfOfRequiredCollateral = arith_uint256(totalCollaterals)*100 < (arith_uint256(collateralsLoans.totalLoans) * ratio / 2);
+        auto isDFIHalfOfRequiredCollateral = arith_uint256(totalCollateralsDFI)*100 < (arith_uint256(collateralsLoans.totalLoans) * ratio / 2);
```

Comment:
its called "isHalf" but its true if its NOT half of all collateral

Reply:
Renamed all variables